packages/block-library/src/gallery/test/index.native.js

```diff
@@ -618,3 +629,39 @@ describe( 'Gallery block', () => {
 		expect( getEditorHtml() ).toMatchSnapshot();
 	} );
 } );
+
+describe( 'Gallery block V1 deprecation', () => {
+	beforeAll( () => {
+		// Register all core blocks
+		registerCoreBlocks( {
+			galleryWithImageBlocks: false,
+		} );
+	} );
+
+	afterAll( () => {
+		// Clean up registered blocks
+		getBlockTypes().forEach( ( block ) => {
+			unregisterBlockType( block.name );
+		} );
+	} );
+
+	it( 'shows the Gallery block as Unsupported if the flag galleryWithImageBlocks is set to false', async () => {
+		// Initialize with a gallery that contains one item
+		const screen = await initializeEditor( {
+			initialHtml: `<!-- wp:gallery {"ids":[13,15],"linkTo":"none"} -->
+			<figure class="wp-block-gallery columns-2 is-cropped">
+			<ul class="blocks-gallery-grid"><li class="blocks-gallery-item">
+			<figure><img src="https://test-site.files.wordpress.com/local-image-1.jpeg" data-id="13" class="wp-image-13"/>
+			</figure></li><li class="blocks-gallery-item"><figure>
+			<img src="https://test-site.files.wordpress.com/local-image-2.jpeg" data-id="15" class="wp-image-15"/>
+			</figure></li></ul></figure>
+			<!-- /wp:gallery -->`,
+		} );
+
+		// Get block
+		const unsupportedBlock = await getBlock( screen, 'Unsupported' );
+		expect( unsupportedBlock ).toBeVisible();
+
+		expect( getEditorHtml() ).toMatchSnapshot();
```

Comment:
I'm wondering if we expect the HTML to be different, even though the block is unsupported the HTML output will remain the same, is this accurate? If so, we might not need this expectation.